src/stirling/source_connectors/socket_tracer/bcc_bpf/socket_trace.c

```diff
@@ -372,10 +381,10 @@ static __inline void submit_new_conn(struct pt_regs* ctx, uint32_t tgid, int32_t
                                      enum endpoint_role_t role, enum source_function_t source_fn) {
   struct conn_info_t conn_info = {};
   init_conn_info(tgid, fd, &conn_info);
-  if (addr != NULL) {
-    conn_info.addr = *((union sockaddr_t*)addr);
-  } else if (socket != NULL) {
+  if (socket != NULL) {
     read_sockaddr_kernel(&conn_info, socket);
+  } else if (addr != NULL) {
+    conn_info.raddr = *((union sockaddr_t*)addr);
   }
```

Comment:
Does this reordering indicate that there was a bug with our previous implementation? I believe I understand that this wouldn't work with the new logic, but it seems like our previous code would be susceptible as well.

Reply:
@benkilimnik any thoughts on this.

Reply:
Hey @ddelnano @JamesMBartlett thanks for the review and apologies for the delayed response!

For `accept` syscalls via 
```
process_syscall_accept -> submit_new_conn(args->addr, args->sock_alloc_socket...)
```
our previous logic only looked at the kernel vars when the IP of the connecting client `addr` is `NULL`. Since we only cared about finding the remote address, there was no need to go looking in the socket if we already got it from the args.

The reordering in this PR has us look at the network socket to find the local address even when the remote address is provided.

Note that for `connect` syscalls via 
```
process_syscall_connect -> submit_new_conn(args->addr, /*socket*/ NULL...)
```
we never set the local address from kernel vars because `socket` is always `NULL`